src/blssignature.cpp

```diff
@@ -68,13 +64,12 @@ bool BLSInsecureSignature::Verify(const uint8_t* msg, size_t len, const BLSPubli
 }
 
 bool BLSInsecureSignature::VerifyHash(const uint8_t* hash, const BLSPublicKey& pubKey) const {
-    relic::g1_t pk, gen;
+    relic::g1_t gen;
     relic::g2_t mappedHash;
     relic::gt_t e1, e2;
     g1_get_gen(gen);
-    pubKey.GetPoint(pk);
     g2_map(mappedHash, hash, BLS::MESSAGE_HASH_LEN, 0);
-    pc_map(e1, pk, mappedHash);
+    pc_map(e1, *(relic::g1_t*)&pubKey.q, mappedHash);
```

Comment:
Why are we using . notation instead of a getter?

Reply:
I wanted to get rid of the getters completely so that the internals are fully hidden in the 3 classes (keys and sig). The Aggregate/Mul methods helped a lot with getting rid of them, as the external operations on group elements were the main reason the getters were needed.

Reply:
These pairings are not being done in the most efficient way. pc_map_sim should be used instead, look at VerifyNative. In fact, I think we should remove these methods (VerifyInsecure and VerifyInsecureHas) as VerifyInsecureAggregated does both of them. Also, can we rename it to just Verify? So there is a BLSInsecureSignature::Verify  and BLSSIgnature::Verify.